primitiv/c/device.h

```diff
@@ -10,16 +10,37 @@
 extern "C" {
 #endif
 
+/**
+ * Opaque type of Device.
+ */
 typedef struct primitiv_Device primitiv_Device;
 
-CAPI extern primitiv_Device *primitiv_Device_get_default();
-CAPI extern primitiv_Device *safe_primitiv_Device_get_default(
-    primitiv_Status *status);
-
-CAPI extern void primitiv_Device_set_default(
-    primitiv_Device *device);
-CAPI extern void safe_primitiv_Device_set_default(primitiv_Device *device,
-                                                  primitiv_Status *status);
+/**
+ * Retrieves the current default device.
+ * @param device Pointer to receive the current default device.
+ * @return Status code.
+ */
+CAPI extern primitiv_Status primitiv_Device_get_default(
```

Comment:
Add remark: The pointer returned by this function is owned by the library, and should not be passed to `primitiv_Device_delete()`.